lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8682,10 +8682,20 @@ void ClangImporter::Implementation::importAttributes(
     if (method->isDirectMethod() && !AnyUnavailable) {
       assert(isa<AbstractFunctionDecl>(MappedDecl) &&
              "objc_direct declarations are expected to be an AbstractFunctionDecl");
-      MappedDecl->getAttrs().add(new (C) FinalAttr(/*IsImplicit=*/true));
-      if (auto accessorDecl = dyn_cast<AccessorDecl>(MappedDecl)) {
-        auto attr = new (C) FinalAttr(/*isImplicit=*/true);
-        accessorDecl->getStorage()->getAttrs().add(attr);
+      if (isa<ConstructorDecl>(MappedDecl)) {
+        // TODO: Teach Swift how to directly call these functions.
+        auto attr = AvailableAttr::createPlatformAgnostic(
+            C, "this Objective-C constructor has the objc_direct attribute",
```

Comment:
Although we call them "constructors" in the compiler implementation, we call them "initializers" to users. We might also want to state the rule more clearly:

```
"Swift cannot call Objective-C initializers marked with 'objc_direct'"
```